x/ecocredit/basket/utils.go

```diff
@@ -1,15 +1,11 @@
 package basket
 
 import (
-	"fmt"
-	"regexp"
-
 	"github.com/regen-network/regen-ledger/x/ecocredit/core"
 )
 
 var (
 	RegexBasketDenom = `[a-zA-Z][a-zA-Z0-9/:._-]{2,127}`
-	regexBasketDenom = regexp.MustCompile(fmt.Sprintf(`^%s$`, RegexBasketDenom))
```

Comment:
I think I previously added this in because we do this for regex in core utils and we use the private regex for validation functions but we don't need a validation function for basket denom so it makes sense to remove it here.

```
	RegexClassId      = `[A-Z]{1,3}[0-9]{2,}`
	RegexProjectId    = fmt.Sprintf(`%s-[A-Z0-9]{2,}`, RegexClassId)
	RegexBatchDenom   = fmt.Sprintf(`%s-[0-9]{8}-[0-9]{8}-[0-9]{3,}`, RegexProjectId)
	RegexJurisdiction = `([A-Z]{2})(?:-([A-Z0-9]{1,3})(?: ([a-zA-Z0-9 \-]{1,64}))?)?`

	regexClassId      = regexp.MustCompile(fmt.Sprintf(`^%s$`, RegexClassId))
	regexProjectId    = regexp.MustCompile(fmt.Sprintf(`^%s$`, RegexProjectId))
	regexBatchDenom   = regexp.MustCompile(fmt.Sprintf(`^%s$`, RegexBatchDenom))
	regexJurisdiction = regexp.MustCompile(fmt.Sprintf(`^%s$`, RegexJurisdiction))
```

Reply:
On second thought, we probably should have a validation function specifically for basket denom so that we can catch an invalid basket denom prior to executing the message. We will need to add this back in when we do, so not sure if you want to add the validation function in this pull request or close this and we can add it in a followup. Opened #1163